File: src/main/java/it/contrader/model/Cure.java
package it.contrader.model;

import javax.persistence.Entity;

import javax.persistence.*;
import javax.persistence.GenerationType;
import javax.persistence.Id;


import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

/**
 * 
 * Model dell'entità User. Contiene l'enum che definisce gli usertype (salvati come interi
 * a partire da 0 sul db).
 * 
 * @author Vittorio Valent & Girolamo Murdaca
 * 
 * @see UserDTO
 */
@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
public class Cure {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private long id;
	
	@Column
	private String cf;
	
	private String age;
	
	private String pato;
	
	private String name;
	
	private String quantity;
	
	private String notes;

	
	
}

File: src/models/Cure.ts
export class Cure {
    id: number;
    
    cf: string;
	
	age: number;
	
	pato: string;
	
	name: string;
	
	quantity: string;
	
    notes: string;


    constructor(id: number, cf: string, age: number, pato: string, name: string, quantity: string, notes:string) {
        this.id = id;
        this.cf = cf;
        this.age = age;
        this.pato = pato;
        this.name = name;
        this.quantity = quantity;
        this.notes = notes;
    }   
}